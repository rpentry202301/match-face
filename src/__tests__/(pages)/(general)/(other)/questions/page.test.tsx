import QuestionsPage from "@/app/(pages)/(general)/(other)/questions/page";
import { render, screen, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom";
import { mockData } from "./mock";

// スナップショットテスト
describe("スナップショットテスト", () => {
  global.fetch = jest.fn().mockImplementation((url, config) => {
    if (url === "http://localhost:3000/api/questions") {
      return {
        ok: true,
        json: async () => mockData,
      };
    }
  });
  it("スナップショット", async () => {
    const { container } = render(<QuestionsPage />);
    await waitFor(() => {
      expect(container).toMatchSnapshot();
    });
  });
});

// 機能・インタラクションテスト
describe("一般ユーザー質問一覧画面", () => {
  global.fetch = jest.fn().mockImplementation((url, config) => {
    if (url === "http://localhost:3000/api/questions") {
      return {
        ok: true,
        json: async () => mockData,
      };
      // }
    }
  });
  beforeEach(async () => {
    await waitFor(() => {
      render(<QuestionsPage />);
    });
  });
  describe("テーブル", () => {
    it("取得データの要素数の<button>が存在する", async () => {
      // const dataNumbers = mockData.answerRequestList.length;
      let element: HTMLElement[];
      await waitFor(() => {
        element = screen.getAllByRole("button");
      });
      screen.debug();
      await waitFor(() => expect(element).toHaveLength(3));
    });
  });
  describe("ボタン", () => {
    it("回答済みであれば、ボタンの色が緑である&&「確認する」", async () => {
      let element;
      await waitFor(() => {
        element = screen.getByTestId(
          `confirmButton${mockData.answerRequestList[0].id}`
        );
      });
      console.log(element);
      expect(element).toHaveClass("bg-green");
      expect(element).toHaveTextContent("確認する");
    });
    it("未回答であれば、ボタンの色が赤である&&「回答する」", async () => {
      let element;
      await waitFor(() => {
        element = screen.getByTestId(
          `confirmButton${mockData.answerRequestList[2].id}`
        );
      });
      screen.debug();
      expect(element).toHaveClass("bg-red");
      expect(element).toHaveTextContent("回答する");
    });
    it("確認するボタンにresult/[id]へのLinkがある", async () => {
      let element;
      await waitFor(() => {
        element = screen.getByTestId(
          `linkButton${mockData.answerRequestList[0].id}`
        );
      });
      screen.debug();
      expect(element).toHaveAttribute(
        "href",
        `result/${mockData.answerRequestList[0].id}`
      );
    });
  });
});
