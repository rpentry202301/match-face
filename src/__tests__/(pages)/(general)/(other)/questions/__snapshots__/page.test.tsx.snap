// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`スナップショットテスト スナップショット 1`] = `
<div>
  <div
    class="flex flex-col items-center h-screen "
  >
    <table
      class="table-auto border border-collapse my-20 w-3/4"
    >
      <thead>
        <tr>
          <th
            class="border p-4 bg-neutral-300"
          >
            回答期日
          </th>
          <th
            class="border  bg-neutral-300"
          >
            案件名
          </th>
          <th
            class="border  bg-neutral-300"
          >
            案件概要
          </th>
          <th
            class="border  bg-neutral-300"
          />
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;
